pkg/apis/core/validation/validation.go

```diff
@@ -3181,19 +3202,41 @@ func validateInitContainers(containers []core.Container, regularContainers []cor
 		// Check for port conflicts in init containers individually since init containers run one-by-one.
 		allErrs = append(allErrs, checkHostPortConflicts([]core.Container{ctr}, fldPath)...)
 
-		// These fields are disallowed for init containers.
-		if ctr.Lifecycle != nil {
-			allErrs = append(allErrs, field.Forbidden(idxPath.Child("lifecycle"), "may not be set for init containers"))
-		}
-		if ctr.LivenessProbe != nil {
-			allErrs = append(allErrs, field.Forbidden(idxPath.Child("livenessProbe"), "may not be set for init containers"))
-		}
-		if ctr.ReadinessProbe != nil {
-			allErrs = append(allErrs, field.Forbidden(idxPath.Child("readinessProbe"), "may not be set for init containers"))
-		}
-		if ctr.StartupProbe != nil {
-			allErrs = append(allErrs, field.Forbidden(idxPath.Child("startupProbe"), "may not be set for init containers"))
+		switch {
+		case restartAlways:
+			// TODO: Allow restartable init containers to have a lifecycle hook.
+			if ctr.Lifecycle != nil {
+				allErrs = append(allErrs, field.Forbidden(idxPath.Child("lifecycle"), "may not be set for init containers"))
```

Comment:
I would change this message to be `may not be set for init containers with the 'Always' restartPolicy`

Reply:
actually, it does not change the validation behavior **_YET_**, as a regular init container cannot have `lifecycle` field as well.

I will submit [a follow-up PR](https://github.com/SergeyKanzhelev/kubernetes/pull/22) to change restartable init containers to be able to have `lifecycle` as soon as this PR is merged.